third_party/move/move-compiler-v2/tests/simplifier/simplifier_test4.exp

```diff
@@ -42,6 +42,17 @@ module 0x8675309::M {
 
 
 Diagnostics:
+error: cannot immutably borrow since mutable references exist
+   ┌─ tests/simplifier/simplifier_test4.move:11:17
+   │
+ 8 │         let f = { x = x + 1; &mut ({x = x + 1; z; s}).f };
+   │                              -------------------------- previous mutable field borrow
+   ·
+11 │             foo(freeze(s), { *({x = x + 1; f}) = 0; 1 })
+   │                 ^^^^^^^^^                  - requirement enforced here
```

Comment:
If you look closer, its really two errors. For the first the primary report is on `f`, for the 2nd the primary error is on the entire assignment. These additional errors are a bit annoying since they feel like redundant (even though slightly different), but they are an artifact of that we have now a new rule checked for v1 compliance which has in some cases an overlap with an already existing rule. This happens in multiple .exp files. Sorting this out and avoiding it looks like a larger effort and certainly will make the code less readable. I think its not that bad overall, so staying with it.